src/Bridge/Doctrine/EventListener/PublishMercureUpdatesListener.php

```diff
@@ -0,0 +1,175 @@
+use Symfony\Component\Serializer\SerializerInterface;
+
+/**
+ * Publish resources updates to the Mercure hub.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ *
+ * @experimental
+ */
+final class PublishMercureUpdatesListener
+{
+    use ClassInfoTrait;
+
+    private $resourceClassResolver;
+    private $iriConverter;
+    private $resourceMetadataFactory;
+    private $serializer;
+    private $messageBus;
+    private $publisher;
+    private $expressionLanguage;
+    private $createdEntities;
+    private $updatedEntities;
+    private $deletedEntities;
+
+    public function __construct(ResourceClassResolverInterface $resourceClassResolver, IriConverterInterface $iriConverter, ResourceMetadataFactoryInterface $resourceMetadataFactory, SerializerInterface $serializer, MessageBusInterface $messageBus = null, callable $publisher = null, ExpressionLanguage $expressionLanguage = null)
+    {
+        if (null === $messageBus && null === $publisher) {
+            throw new InvalidArgumentException('A message bus or a publisher must be provided.');
+        }
+
+        $this->resourceClassResolver = $resourceClassResolver;
+        $this->iriConverter = $iriConverter;
+        $this->resourceMetadataFactory = $resourceMetadataFactory;
+        $this->serializer = $serializer;
+        $this->messageBus = $messageBus;
+        $this->publisher = $publisher;
+        $this->expressionLanguage = $expressionLanguage ?? class_exists(ExpressionLanguage::class) ? new ExpressionLanguage() : null;
+        $this->reset();
+    }
+
+    /**
+     * Collects created, updated and deleted entities.
+     */
+    public function onFlush(OnFlushEventArgs $eventArgs)
+    {
+        $uow = $eventArgs->getEntityManager()->getUnitOfWork();
+
+        foreach ($uow->getScheduledEntityInsertions() as $entity) {
+            $this->storeEntityToPublish($entity, 'createdEntities');
+        }
+
+        foreach ($uow->getScheduledEntityUpdates() as $entity) {
+            $this->storeEntityToPublish($entity, 'updatedEntities');
+        }
+
+        foreach ($uow->getScheduledEntityDeletions() as $entity) {
+            $this->storeEntityToPublish($entity, 'deletedEntities');
+        }
+    }
+
+    /**
+     * Purges tags collected during this request, and clears the tag list.
+     */
+    public function postFlush()
+    {
+        try {
+            foreach ($this->createdEntities as $entity) {
+                $this->publishUpdate($entity, $this->createdEntities[$entity]);
+            }
+
+            foreach ($this->updatedEntities as $entity) {
+                $this->publishUpdate($entity, $this->updatedEntities[$entity]);
+            }
+
+            foreach ($this->deletedEntities as $iri => $targets) {
+                $this->publishUpdate($iri, $targets);
+            }
+        } finally {
+            $this->reset();
+        }
+    }
+
+    private function reset(): void
+    {
+        $this->createdEntities = new \SplObjectStorage();
```

Comment:
is this the same as `removeAll`?

Reply:
`removeAll` takes a mandatory argument as parameter.

Reply:
No idea :) I would keep it as is for now because it's "safer", but if your suggestion improve GC or something similar, let me know and I'll update!